components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/listener/ConsoleRoleListener.java

```diff
@@ -0,0 +1,256 @@
+import static org.wso2.carbon.identity.api.resource.collection.mgt.constant.APIResourceCollectionManagementConstants.APIResourceCollectionConfigBuilderConstants.VIEW_ACTION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.APPLICATION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.CONSOLE_APP_AUDIENCE_NAME;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.CONSOLE_ORG_SCOPE_PREFIX;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.CONSOLE_SCOPE_PREFIX;
+
+/**
+ * Console role listener to populate organization console application roles permissions.
+ */
+public class ConsoleRoleListener extends AbstractRoleManagementListener {
+
+    @Override
+    public int getDefaultOrderId() {
+
+        return 4;
+    }
+
+    @Override
+    public boolean isEnable() {
+
+        return true;
+    }
+
+    @Override
+    public void preAddRole(String roleName, List<String> userList, List<String> groupList, List<Permission> permissions,
+                           String audience, String audienceId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        if (APPLICATION.equals(audience)) {
+            List<Permission> consoleFeaturePermissions = getConsoleFeaturePermissions(permissions);
+            if (ArrayUtils.isNotEmpty(consoleFeaturePermissions.toArray())) {
+                // If console features are added to the role, then we need to we only need to persist the console
+                // permissions.
+                permissions.retainAll(consoleFeaturePermissions);
+            }
+        }
+    }
+
+    @Override
+    public void postGetRole(Role role, String roleId, String tenantDomain) throws IdentityRoleManagementException {
+
+
+        if (isConsoleRole(roleId, tenantDomain)) {
+            // Get updated console role permissions with newly added read and write scopes from API resource collection.
+            List<Permission> rolePermissions = getUpdatedPermissions(role.getPermissions(), tenantDomain);
+            role.setPermissions(rolePermissions);
+        }
+    }
+
+    @Override
+    public void postGetPermissionListOfRole(List<Permission> permissionListOfRole, String roleId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        if (isConsoleRole(roleId, tenantDomain)) {
+            List<Permission> rolePermissions = getUpdatedPermissions(permissionListOfRole, tenantDomain);
+            permissionListOfRole.retainAll(rolePermissions);
+        }
+    }
+
+    @Override
+    public void postGetPermissionListOfRoles(List<String> permissions, List<String> roleIds, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        boolean isConsoleRole = false;
+        for (String roleId : roleIds) {
+            if (isConsoleRole(roleId, tenantDomain)) {
+                isConsoleRole = true;
+                break;
+            }
+        }
```

Comment:
By looking at the permissions also we can decide a console role. A console role exist permissions starts with `console:`